src/thunderbots/software/radio_communication/mrf/annunciator.h

```diff
@@ -16,26 +16,26 @@ class Annunciator
     explicit Annunciator(ros::NodeHandle& node_handle);
 
     /**
-     * Handles diagnostics for each robot.
+     * Handles diagnostics and messages for each robot.
+     * Returns a boolean if there were new messages since the last status update.
      *
      * @param index Robot number.
      * @param data The data of the status packet.
      * @param len The length of the packet.
      * @param lqi Link quality.
      * @param rssi Received signal strength indicator.
      *
-     * @return the fully-constructed RobotStatus that was published
+     * @return true if new messages since last status update
      */
-    thunderbots_msgs::RobotStatus handle_robot_message(int index, const void* data,
-                                                       std::size_t len, uint8_t lqi,
-                                                       uint8_t rssi);
+    bool handle_robot_message(int index, const void* data, std::size_t len, uint8_t lqi,
+                              uint8_t rssi);
```

Comment:
Not a blocking comment, but it would be preferable if this still returned the status, and then we published the ROS message from the top level. This would help keep radio isolated from ROS, which would be a nice bit of encapsulation to maintain.

Reply:
Yeah you do have a valid point. It's just that this function gets called by a callback in `MRFDongle`, which is an even less appropriate place to have ROS messages. Also, if I were to have this function return the status then that means `MRFDongle` will have to be the class that maintains message state.

Reply:
I mean, it would make sense to me to have the `MRFDongle` maintain the robot state (or at least as much sense as it does to have it maintained in the Annunciator ¯\_(ツ)_/¯), but don't worry about it if it's a huge hassle.